internal/core/src/query/visitors/ExecExprVisitor.cpp

```diff
@@ -253,14 +321,14 @@ ExecExprVisitor::ExecDataRangeVisitorImpl(FieldId field_id,
         auto& indexing =
             segment_.chunk_scalar_index<IndexInnerType>(field_id, chunk_id);
         auto this_size = const_cast<Index*>(&indexing)->Count();
-        BitsetType result(this_size);
+        FixedVector<bool> result(this_size);
         for (int offset = 0; offset < this_size; ++offset) {
             result[offset] = index_func(const_cast<Index*>(&indexing), offset);
         }
         results.emplace_back(std::move(result));
     }
 
-    auto final_result = Assemble(results);
+    auto final_result = AssembleChunk(results);
```

Comment:
Anyway to skip this assembling if results length is 1

Reply:
Results are bool arrays but final result is bitset, so must convert it even one result.